backend/controllers/user-controller.js

```diff
@@ -38,6 +46,12 @@ exports.signup = (req, res) => {
 };
 
 exports.signin = (req, res) => {
+    googleRecaptcha.verify({ response: req.body.ReCAPTCHAValue }, (error) => {
+        if (error) {
+            res.status(HttpStatus.BAD_REQUEST).json({ message: 'You are not human' });
+        }
+    });
+
     User.findOne({ email: req.body.email })
```

Comment:
I think we should move code of creating user to callback of verify method on line:50.